frontend/public/components/modals/configure-ns-pull-secret-modal.jsx

```diff
@@ -152,17 +152,33 @@ class ConfigureNamespacePullSecret extends PromiseComponent {
       promise = k8sPatch(SecretModel, pullSecret, patch);
     } else {
       const data = {};
+      const pullSecretName = event.target.elements['namespace-pull-secret-name'].value;
       data[CONST.PULL_SECRET_DATA] = secretData;
 
       const secret = {
         metadata: {
-          name: event.target.elements['namespace-pull-secret-name'].value,
+          name: pullSecretName,
           namespace: namespace.metadata.name,
         },
         data,
         type: CONST.PULL_SECRET_TYPE,
       };
-      promise = k8sCreate(SecretModel, secret);
+      const defaultServiceAccount = {
```

Comment:
We should add a version of `k8sPatch` that takes in name and namespace params instead of creating a fake resource.